release/tar/linux/opensearch-dashboards-tar-build.sh

```diff
@@ -0,0 +1,147 @@
+    echo "error: Please specify the version in the manifest file"
+    usage
+    exit 1
+fi
+
+if [ "$SCHEMA_VERSION" != "1.0" ]; then
+    echo "error: This script only supports manifest schema version 1.0. Please use the manifest file with right schema version"
+    usage
+    exit 1
+fi
+
+if [ "$PLATFORM" != "linux" ]; then
+    echo "error: $PLATFORM platform is not supported yet! Only supported platform is linux"
+    usage
+    exit 1
+fi
+
+if [ "$ARCHITECTURE" != "x64" ] && [ "$ARCHITECTURE" != "arm64" ]
+then
+    echo "error: $ARCHITECTURE architecture is not supported yet! Only supported architectures are x64 and arm64"
+    usage
+    exit 1
+fi
+
+if [[ ( "$ARCHITECTURE" == "arm64" && ( "$UNDERLYING_ARCH" != "aarch64" && "$UNDERLYING_ARCH" != "arm64" )) ||  ( "$ARCHITECTURE" == "x64" && ( "$UNDERLYING_ARCH" != "x86_64" && "$UNDERLYING_ARCH" != "amd64" )) ]]
+then
+    echo "error: The underlying architecture($UNDERLYING_ARCH) doesnot match the product architecture($ARCHITECTURE) that you are trying to build"
+    echo "Please use the appropriate build environment"
+    exit 1
+fi
+
+REPO_ROOT=`git rev-parse --show-toplevel`
+ROOT=`dirname $(realpath $0)`; echo $ROOT; cd $ROOT
+DIR_NAME="opensearch-dashboards-$VERSION"
+WORKING_DIR="$ROOT/$DIR_NAME"
+TARGET_DIR="$ROOT/target"
+PLUGINS_TEMP=`mktemp -d`
+OPENSEARCH_DASHBOARDS_CORE_URL=`yq eval '.products.opensearch-dashboards.opensearch-dashboards-min' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+OPENSEARCH_DASHBOARDS_PLUGINS_URLS=`yq eval '.products.opensearch-dashboards.plugins' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+
+# # Download plugins to local 
+mkdir $WORKING_DIR
+mkdir -p $TARGET_DIR
+echo "working dir is: ${WORKING_DIR}"
+IFS=$'\n'
+for plugin in ${OPENSEARCH_DASHBOARDS_PLUGINS_URLS}; do
+  aws s3 cp $plugin $PLUGINS_TEMP/
+done
+
+# Download OpenSearch Dashboards min tarball
+wget -q ${OPENSEARCH_DASHBOARDS_CORE_URL} -O opensearch-dashboards.tar.gz
+tar -xzf opensearch-dashboards.tar.gz --strip-components=1 --directory "$WORKING_DIR" && rm -rf opensearch-dashboards.tar.gz
+
+# Install plugins to OpenSearch
+for plugins in $OPENSEARCH_DASHBOARDS_PLUGINS_URLS; do
+    plugin=`basename $plugins`
+    $WORKING_DIR/bin/opensearch-dashboards-plugin --allow-root install file:$PLUGINS_TEMP/$plugin
+done
+
+echo "Plugins installed are:"
+ls -tlr $WORKING_DIR/plugins
+
+# Copy the tarball installation script 
+cp $REPO_ROOT/config/opensearch_dashboards.yml $WORKING_DIR/config/
+
+# Tar the bundle and clean up
+rm -rf $PLUGINS_TEMP
+cd $ROOT
+echo "Generating the tarball"
+tar -czf $TARGET_DIR/opensearch-dashboards-$VERSION-$PLATFORM-$ARCHITECTURE.tar.gz $DIR_NAME
+cd $TARGET_DIR
+shasum -a 512 opensearch-dashboards-$VERSION-$PLATFORM-$ARCHITECTURE.tar.gz > opensearch-dashboards-$VERSION-$PLATFORM-$ARCHITECTURE.tar.gz.sha512
+shasum -a 512 -c opensearch-dashboards-$VERSION-$PLATFORM-$ARCHITECTURE.tar.gz.sha512
+rm -rf ${WORKING_DIR}
```

Comment:
You dont rm TARGET_DIR as well?

Reply:
Suggest the cleanup should happen earlier, as well as later in the script.

Reply:
Target contains the final artifact and shasum we cannot remove that